commands/scp.go

```diff
@@ -0,0 +1,136 @@
+package commands
+
+import (
+	"errors"
+	"fmt"
+	"os"
+	"os/exec"
+	"strings"
+
+	"github.com/codegangsta/cli"
+	"github.com/docker/machine/libmachine"
+	"github.com/docker/machine/log"
+)
+
+var (
+	ErrMalformedInput = fmt.Errorf("The input was malformed")
+)
+
+var (
+	// TODO: possibly move this to ssh package
+	baseSSHOptions = []string{
+		"-o", "IdentitiesOnly=yes",
+		"-o", "StrictHostKeyChecking=no",
+		"-o", "UserKnownHostsFile=/dev/null",
+		"-o", "LogLevel=quiet", // suppress "Warning: Permanently added '[localhost]:2022' (ECDSA) to the list of known hosts."
+	}
+)
+
+func getInfoForScpArg(hostAndPath string, mcn libmachine.Machine) (*libmachine.Host, string, []string, error) {
```

Comment:
Well, I did figure it out just fine when I read the full function. And I can't speak for what the average coder does, but when I'm reading code I generally read a function's signature first to try to figure out what it's doing, then I look for a `return` at the end of the function. In this case it's `return nil, "", nil, ErrMalformedInput`. This is a super-minor thing, and I'm picking nits, but I do think it's nice to make code as easy to read as possible :)